physical/postgresql/configure_test.go

```diff
@@ -0,0 +1,65 @@
+package postgresql
+
+import (
+	"os"
+	"testing"
+)
+
+func TestConnectionURL(t *testing.T) {
+	type input struct {
+		envar string
+		conf  map[string]string
+	}
+
+	var cases = []struct {
+		name  string
+		want  string
+		input input
```

Comment:
Nitpick, and not required: instead of a `[]struct` with a `name` attribute, you could use a `map[string]struct` and the map key becomes the name. Here's an example https://github.com/hashicorp/vault/blob/master/command/agent/template/template_test.go#L51-L63

A subtle, hidden benefit to that is when ranging over a map there is no guaranteed order, so it could avoid a situation were tests pass because of the setup from previous scenarios (though I don't believe that's possible in this specific case).

Just a suggestion 😄

Reply:
@catsby another great suggestion! Anything to avoid flaky tests.

Reply:
@catsby actually, this change caught flakiness. Basically, 
```golang
defer os.Setenv("PG_CONNECTION_URL", "")
```
is not proper cleanup because `os.LookupEnv` still returns `ok == true`.

The only option (that I am aware of) to actually delete environment variables is [`os.Clearenv`](https://golang.org/pkg/os/#Clearenv). While this _should_ be fine assuming that there are no test dependencies elsewhere, where envs get set in a setup function or something and used in a test, I would rather not use it.

I think this can be avoided by only overwriting the `conf` value if the env value `!= ""`

Reply:
There is [os.Unsetenv](https://golang.org/pkg/os/#Unsetenv)

Reply:
I can't believe I didn't see this. Thanks!